app/components/Review/widgets/FileWidget.tsx

```diff
@@ -53,24 +53,29 @@ const FileWidget: React.FC<WidgetProps> = ({
   const filesArray = useMemo(() => {
     return value || [];
   }, [value]);
-
+  const filesWithIcon = [];
   return (
     <div>
-      {rfiList?.map((rfi, i) => {
+      {rfiList?.map((rfi) => {
         // loop through the rfis and check if there are any files for the current field
         const rfiFiles = rfi.jsonData.rfiAdditionalFiles;
         const isFileField =
           rfiFiles && Object.keys(rfiFiles).includes(fieldName);
         if (!isFileField) return null;
-
+        const fileFields = rfiFiles[fieldName] || [];
+        // Sorting the file fields by id, latest file will always have higher id
+        // not sorting by uploadedAt as it is not available for all files
+        const sortedFileFields = fileFields.slice().sort((a, b) => b.id - a.id);
         const attachments = rfi?.attachments?.nodes;
 
         return (
           <StyledFile key={rfi.id}>
-            {rfiFiles[fieldName]?.map((el, index) => {
+            {sortedFileFields?.map((el, index) => {
               // loop through the list of files for the current field
               const isSingleFile = !options?.allowMultipleFiles;
-              const isDisplayIcon = isSingleFile && i === 0;
+              const isDisplayIcon =
```

Comment:
just wondering why we push to `filesWithIcon` array, instead can't we use  `const isDisplayIcon = isSingleFile && i === 0;` ?

Reply:
because the same file might there for multiple RFIs, so the intention is to track which types of files we have added the star to, otherwise it will duplicate stars by RFI, this is what happens: 

<img width="567" alt="image" src="https://github.com/bcgov/CONN-CCBC-portal/assets/61289255/948e6c7a-2d77-43b9-b3bd-cb854d2aa20b">

Reply:
Thanks Rafael, Got it, was thinking of rfi categories, not accounting multiple raised rfis.